extension/src/lib/captionsEnactor.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+// The contents of this file are written in plain JavaScript to avoid conflicts of duplicate dependencies (e.g.
+// Fluid Infusion, jQuery, and etc.) when injected into arbitrary web pages.
+
+var gpii = gpii || {};
+
+(function (gpii) {
+
+    gpii.uioPlus = {
+        guid: 0
+    };
+
+    gpii.uioPlus.allocateSimpleId = function (element) {
+        if (!element.id) {
+            var simpleId = "gpii-uioPlus-id-" + Date.now() + "-" + gpii.uioPlus.guid++;
```

Comment:
Why do we add the date here in addition to the guid? The guid itself is already unique enough with respect to successive execution. If you're worried about uniqueness across iframes, I suggest you cache some string as a variable in this file's scope, e.g. using a strategy like this: https://stackoverflow.com/questions/7200722/jquery-expando-properties?lq=1 - e.g. the string can be  (Math.random() ).replace( /\D/g, "")